temporalio/worker/_workflow_instance.py

```diff
@@ -1999,6 +2077,101 @@ def _convert_payloads(
                 raise
             raise RuntimeError("Failed decoding arguments") from err
 
+    def _converters_with_context(
+        self,
+        context: temporalio.converter.SerializationContext,
+    ) -> Tuple[
+        temporalio.converter.PayloadConverter,
+        temporalio.converter.FailureConverter,
+    ]:
+        """Construct workflow payload and failure converters with the given context.
+
+        This plays a similar role to DataConverter._with_context, but operates on PayloadConverter
+        and FailureConverter only (since payload encoding/decoding is done by the worker, outside
+        the workflow sandbox).
+        """
+        payload_converter = self._context_free_payload_converter
+        failure_converter = self._context_free_failure_converter
+        if isinstance(payload_converter, temporalio.converter.WithSerializationContext):
+            payload_converter = payload_converter.with_context(context)
+        if isinstance(failure_converter, temporalio.converter.WithSerializationContext):
+            failure_converter = failure_converter.with_context(context)
+        return payload_converter, failure_converter
+
+    def get_serialization_context(
+        self,
+        command_info: Optional[_command_aware_visitor.CommandInfo],
+    ) -> Optional[temporalio.converter.SerializationContext]:
+        workflow_context = temporalio.converter.WorkflowSerializationContext(
```

Comment:
Many times in this method, this context isn't needed, why instantiate it every time as if it's always needed?